great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -273,18 +273,19 @@ def _init_analytics(self) -> None:
 
     def _determine_analytics_enabled(self) -> bool:
         """
-        In order to determine whether analytics should be enabled, we check two sources:
-          - The `analytics_enabled` key in the GX config file
-            - If missing, we assume True.
+        Determine if analytics are enabled using the following precedence
+          - The `analytics_enabled` key in the GX config
           - The `GX_ANALYTICS_ENABLED` environment variable
-
-        If both are True, analytics will be enabled. If either is False, analytics will be disabled.
+          - Otherwise, assume True
         """
-        config_file_enabled = self.config.analytics_enabled
-        if config_file_enabled is None:
-            config_file_enabled = True
+        config_enabled = self.config.analytics_enabled
         env_var_enabled = ENV_CONFIG.posthog_enabled
-        return config_file_enabled and env_var_enabled
+        if config_enabled is not None:
+            return config_enabled
+        elif env_var_enabled is not None:
+            return env_var_enabled
```

Comment:
This will always be non-null due to Pydantic coercion I think?

Reply:
Looks like it's still marshmallow, and is optional: https://github.com/great-expectations/great_expectations/blob/develop/great_expectations/data_context/types/base.py#L1116

Reply:
oh I see. That's for the environment variable. Yeah, I guess that'd have to change as well if we otherwise want the behavior implied in this method.

Reply:
Updated.